system_server crash while ending a video call (2024121200)

System crash just as I was hanging up a video call with SchildiChat.
```
type: crash

SystemUptimeMs: 48752184
Process: system_server
Timestamp: 2024-12-15 15:00:06.942+0100
Build: google/panther/panther:15/AP4A.241205.013/2024121200:user/release-keys
Crash-Handler: com.android.internal.os.RuntimeInit$KillApplicationHandler
Loading-Progress: 1.0
Dropped-Count: 0

java.lang.RuntimeException: Error receiving broadcast Intent { act=com.android.server.notification.TimeToLiveHelper dat=timeout:/... flg=0x10000010 pkg=android (has extras) } in com.android.server.notification.TimeToLiveHelper$1@878b2d4
	at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0(LoadedApk.java:1827)
	at android.app.LoadedApk$ReceiverDispatcher$Args.$r8$lambda$mcNAAl1SQ4MyJPyDg8TJ2x2h0Rk(Unknown Source:0)
	at android.app.LoadedApk$ReceiverDispatcher$Args$$ExternalSyntheticLambda0.run(D8$$SyntheticClass:0)
	at android.os.Handler.handleCallback(Handler.java:991)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loopOnce(Looper.java:232)
	at android.os.Looper.loop(Looper.java:317)
	at com.android.server.SystemServer.run(SystemServer.java:995)
	at com.android.server.SystemServer.main(SystemServer.java:673)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:591)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:865)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.Object.equals(java.lang.Object)' on a null object reference
	at com.android.server.notification.NotificationManagerService.findNotificationByListLocked(NotificationManagerService.java:11235)
	at com.android.server.notification.NotificationManagerService.findNotificationByKeyLocked(NotificationManagerService.java:11180)
	at com.android.server.notification.NotificationManagerService.-$$Nest$mfindNotificationByKeyLocked(NotificationManagerService.java:0)
	at com.android.server.notification.NotificationManagerService$2.timeoutNotification(NotificationManagerService.java:1855)
	at com.android.server.notification.TimeToLiveHelper$1.onReceive(TimeToLiveHelper.java:191)
	at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0(LoadedApk.java:1819)
	... 11 more
```

--- a/services/core/java/com/android/server/notification/TimeToLiveHelper.java
+++ b/services/core/java/com/android/server/notification/TimeToLiveHelper.java
@@ -188,7 +188,9 @@ public class TimeToLiveHelper {
                         timeoutKey = earliest.second;
                     }
                 }
-                mNm.timeoutNotification(timeoutKey);
+                if (timeoutKey != null) {
+                    mNm.timeoutNotification(timeoutKey);
+                }
             }
         }
     };

